### PR TITLE
Validate ClientHello session_id field length and send alert on failure (1.0.2)

### PR DESCRIPTION
**Pull request by ghedo**
RT#4080

### CODE REVIEW

File: ssl/s2_srvr.c [hunk 1]
```diff
@@ -598,6 +598,11 @@ static int get_client_hello(SSL *s)
         s->s2->tmp.cipher_spec_length = i;
         n2s(p, i);
         s->s2->tmp.session_id_length = i;
+        if ((i < 0) || (i > SSL_MAX_SSL_SESSION_ID_LENGTH)) {
+            ssl2_return_error(s, SSL2_PE_UNDEFINED_ERROR);
+            SSLerr(SSL_F_GET_CLIENT_HELLO, SSL_R_LENGTH_MISMATCH);
+            return (-1);
```
Comment (mattcaswell) [inline ssl/s2_srvr.c]: nit: where ever we write new code with a "return" the expression should not be in parens as per the coding style document.

**Comment by mattcaswell** [comment]
+1, subject to the nit being fixed.

**Comment by ghedo** [comment]
Nit fixed just now.

**Comment by mattcaswell** [comment]
+1

**Comment by richsalz** [comment]
+1 for matt to merge and fix it before commit.

**Comment by mattcaswell** [comment]
Patch applied. Thanks.

**Pull request closed without merging**